src/plugins/intel_gpu/tests/functional/subgraph_tests/dynamic/convert_bf16_fp32_fp16.cpp

```diff
@@ -0,0 +1,346 @@
+// Copyright (C) 2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "common_test_utils/ov_tensor_utils.hpp"
+#include "shared_test_classes/base/ov_subgraph.hpp"
+#include "transformations/rt_info/decompression.hpp"
+
+#include "openvino/op/parameter.hpp"
+#include "openvino/op/constant.hpp"
+#include "openvino/op/result.hpp"
+#include "openvino/op/matmul.hpp"
+#include "openvino/op/reshape.hpp"
+#include "openvino/op/convert.hpp"
+#include "openvino/op/subtract.hpp"
+#include "openvino/op/transpose.hpp"
+
+namespace {
+using ov::test::InputShape;
+
+struct ShapeParams {
+    ShapeParams() = default;
+    ShapeParams(InputShape data_shape, ov::Shape weights_shape, int weights_group_size = -1)
+        : data_shape(std::move(data_shape)),
+          weights_shape(std::move(weights_shape)),
+          weights_group_size(weights_group_size) {}
+
+    InputShape data_shape;
+    ov::Shape weights_shape;
+    // Decompression group size. If the value is equal to -1, ordinary decompression is used
+    int weights_group_size;
+};
+
+using MatmulWeightsDecompressionParams = std::tuple<ShapeParams,              // input shapes
+                                                    ov::element::Type,        // weights type
+                                                    ov::element::Type,        // activations type
+                                                    bool,                     // transpose on weights
+                                                    bool,                     // decompression subtract
+                                                    bool,                     // reshape on decompression constants
+                                                    bool>;                    // per-tensor zero-point
+
+class BF16WeightsDecompression : public testing::WithParamInterface<MatmulWeightsDecompressionParams>,
+                                   virtual public ov::test::SubgraphBaseTest {
+public:
+    ov::CompiledModel cM;
+    ov::CompiledModel cMF16;
+    std::shared_ptr<ov::Model> functionF16;
+    ov::ParameterVector paramsf16{std::make_shared<ov::op::v0::Parameter>(ov::element::bf16, ov::Shape{{2, 2}})};
```

Comment:
What is the target scenario? Shouldn't bf16 data type be applied to constants, not params?

Reply:
done